src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -481,6 +481,12 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     var value;
 
     if (__DEV__) {
+      warning(
+        !(workInProgress.type.prototype &&
+          workInProgress.type.prototype.render),
+        'Warning: Using a class with a render method as a function, ' +
+          'did you forget to extend React.Component?',
```

Comment:
I would tweak the wording to:

>The <%s /> component appears to have a render method, but doesn't extend React.Component. This is likely to cause errors. Change %s to extend React.Component instead.

You’ll need to pass the component name (use `getComponentName`).